### ISSUE TITLE
"extract_features.py" freezes

### ISSUE THREAD

**Issue by meegoStar** [imatge-upc/activitynet-2016-cvprw#7]
Hello, I'm running `extract_features.py` using the command `KERAS_BACKEND=theano python -u extract_features.py -d ~/Videos/videos`. However, the program will freeze and refuse to make any further progress with the latest printed messages only showing
```
Time to fetch [video name] video: [time] seconds
Time to fetch [video name] video: [time] seconds
Time to fetch [video name] video: [time] seconds
```
I could only force `extract_features.py` terminated, and run the command again to continue.
Is it normal that `extract_features.py` freezes?

**Comment by kertansul** [comment]
@meegoStar hi, did you find out the solution?

**Comment by meegoStar** [comment]
No. Unfortunately, we just kept manually restarting the program until all the features been extracted.

**Comment by albertomontesg** [comment]
I experienced the same while running this, and I wasn't able to find the cause (as there is concurrency to fetch the videos, maybe at some point or some video make this program to freeze). I rerun the code until I had all the video features extracted. I'm sorry I can not help more.

**Comment by cyanine-gi** [comment]
Maybe there's a deadlock in the script "extract_feature.py". I'm trying to solve this deadlock if it exists. I'll leave a message here if I succeed.

**Comment by buaa-luzhi** [comment]
Hello,@cyanine-gi 

Have you solved the problem?

Can you give me some advice?

Thank you very much!

**Comment by cyanine-gi** [comment]
Maybe it's caused by the circumstance check of "_stop_all_generators" and "_stop_all_extractors".
Since line 97 don't check the queue again,maybe the extracting_features_task() won't quit.
To solve this, I simply changed the code and set this task to a single thread mode, which is not that fast when processing files but more reliable.

**Comment by buaa-luzhi** [comment]
Hello,@cyanine-gi 
Can you give me samples code? 
Thank you so much!

Best regards

**Comment by Zakia13** [comment]
Hello i'm experiencing the same problem, **extract_features.py** stucks in a while loop of extractor_process and saver_process because generator_process produces empty data queues. i don't know why data_gen_queue remains empty although VideoGenerator class is running successfully.
Kindly help me how to resolve this issue.
Thanks.
